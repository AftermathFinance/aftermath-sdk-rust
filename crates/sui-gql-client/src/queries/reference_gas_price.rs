use super::Error;
use crate::scalars::BigInt;
use crate::{GraphQlClient, GraphQlResponseExt, missing_data, schema};

pub(super) async fn query<C: GraphQlClient>(client: &C) -> Result<u64, Error<C::Error>> {
    let query: Query = client
        .query(())
        .await
        .map_err(Error::Client)?
        .try_into_data()?
        .ok_or(missing_data!("No data in response"))?;

    Ok(query
        .epoch
        .ok_or(missing_data!("Epoch not found"))?
        .reference_gas_price
        .ok_or(missing_data!("Epoch reference gas price"))?
        .into_inner())
}

#[cfg(test)]
#[allow(clippy::unwrap_used)]
#[test]
fn gql_output() {
    use cynic::QueryBuilder as _;

    let operation = Query::build(());
    insta::assert_snapshot!(operation.query, @r###"
    query Query {
      epoch {
        referenceGasPrice
      }
    }
    "###);
}

#[derive(cynic::QueryFragment, Clone, Debug)]
struct Query {
    epoch: Option<Epoch>,
}

#[derive(cynic::QueryFragment, Clone, Debug)]
struct Epoch {
    reference_gas_price: Option<BigInt<u64>>,
}
