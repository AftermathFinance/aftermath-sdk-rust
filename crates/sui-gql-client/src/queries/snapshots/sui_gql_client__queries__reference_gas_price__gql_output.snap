---
source: crates/sui-gql-client/src/queries/reference_gas_price.rs
expression: operation.query
---
query Query {
  epoch {
    referenceGasPrice
  }
}
